### PR TITLE
Add use cases overview and e-commerce example

### PR DESCRIPTION
**Pull request by lesters**
@kkraune Please review.
@jobergum FYI.

Anything missing?

### CODE REVIEW

File: documentation/use-case-shopping.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+this, as feeding the reviews more than once will result in increased values. To
+do this correctly, you should probably calculate this offline so a re-feed does
+not cause unexpected results.
+
+### Features in applications
+
+* [Multiple document types](https://docs.vespa.ai/documentation/search-definitions.html)
+
+    Vespa models data as documents, which are configured in search definitions
+    that defines how documents should be storedm indexed, ranked, and searched.
```
Comment (kkraune) [inline documentation/use-case-shopping.md]: stored

File: documentation/use-case-shopping.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    setup.
+
+* [Custom configuration](https://docs.vespa.ai/documentation/configuring-components.html)
+
+    When creating custom components in Vespa, for instance document processors,
+    searchers or handler, one can use custom configuration to inject config
+    parameters into the components. This involves defining a config definition
+    (a `.def` file), which created a config class. You can instantiate this
+    class with data in `services.xml` and the resulting object is injected to
+    you component during construction. This application uses custom config
```
Comment (kkraune) [inline documentation/use-case-shopping.md]: the component

File: documentation/use-case-shopping.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    to set up the Vespa host details for the handler.
+
+* [Partial update](https://docs.vespa.ai/documentation/reference/document-json-format.html#update)
+
+    With Vepsa you can make changes to an existing document without submitting
```
Comment (kkraune) [inline documentation/use-case-shopping.md]: vespa

File: documentation/use-cases.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+different features with links to relevant documentation, or as a starting point
+for new similar applications.
+
+More applications will be added soon.
+
+#### [E-commerce: shopping and product catalog](/documentation/use-case-shopping.html)
+
+**Highlighted features**:
+
+* Multiple document types
```
Comment (kkraune) [inline documentation/use-cases.md]: multiple (small 'm')

**Comment by lesters** [comment]
Jeez, my spell checker has evidently turned itself off.

**Pull request merged**